### PR TITLE
fixed #1156, socketMode error not being properly thrown

### LINKED ISSUE #1156: [socket-mode] webClient error is swallowed
### Description

When `webClient.apiCall` throws an error it is not propagated and handled by `onFailure`. Instead an error is logged and forgotten.

```ts
              return this.webClient.apiCall('apps.connections.open').then((result: WebAPICallResult) => {
                return result;
              }).catch((error) => {
                this.logger.error(error);
              });
```

As a result machine changes state to `authenticated` handled where `context.result` is `undefined`

Which ultimately leads to `TypeError: Cannot read property 'url' of undefined` in 

```ts
                this.setupWebSocket(context.result.url);
```

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Packages:

Select all that apply:

- [ ] `@slack/web-api`
- [ ] `@slack/events-api`
- [ ] `@slack/interactive-messages`
- [ ] `@slack/rtm-api`
- [ ] `@slack/webhooks`
- [ ] `@slack/oauth`
- [ ] I don't know
- [x] `@slack/socket-mode`

#### Reproducible in:

package version: 1.0.0

node version: v12.18.4

OS version(s): macOS Catalina

#### Steps to reproduce:

1. Provide wrong app token.
2. Start bolt with `socketMode: true`

#### Expected result:

Error is rethrown after logging and state machine does not transition to "authenticated".

#### Actual result:

Error is swallowed and state machine transitions to "authenticated".

#### Attachments:

```
⚡️ Bolt app started
[DEBUG]  web-api:WebClient:0 http response received
[ERROR]  socket-mode:SocketModeClient:0 Error: An API error occurred: not_authed
    at Object.platformErrorFromResult (/Users/#####/node_modules/@slack/socket-mode/node_modules/@slack/web-api/dist/errors.js:51:33)
    at WebClient.apiCall (/Users/#####/node_modules/@slack/socket-mode/node_modules/@slack/web-api/dist/WebClient.js:156:28)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  code: 'slack_webapi_platform_error',
  data: {
    ok: false,
    error: 'not_authed',
    response_metadata: { acceptedScopes: [Array] }
  }
}
[DEBUG]  socket-mode:SocketModeClient:0 transitioning to state: connecting:authenticated
(node:92065) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'url' of undefined
    at /Users/#####/node_modules/@slack/socket-mode/dist/SocketModeClient.js:76:48
    at /Users/#####/node_modules/finity/lib/utils/invokeEach.js:11:15
    at Array.forEach (<anonymous>)
    at invokeEach (/Users/#####/node_modules/finity/lib/utils/invokeEach.js:10:7)
    at StateMachine.enterState (/Users/#####/node_modules/finity/lib/core/StateMachine.js:125:32)
    at StateMachine.executeTransition (/Users/#####/node_modules/finity/lib/core/StateMachine.js:116:12)
    at /Users/#####/node_modules/finity/lib/core/StateMachine.js:285:16
    at TaskScheduler.execute (/Users/#####/node_modules/finity/lib/core/TaskScheduler.js:29:7)
    at StateMachine.executeTrigger (/Users/#####/node_modules/finity/lib/core/StateMachine.js:282:24)
    at StateMachine.handleAsyncActionComplete (/Users/#####/node_modules/finity/lib/core/StateMachine.js:195:10)
(node:92065) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:92065) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[DEBUG]  web-api:WebClient:0 http response received
```

### PR DESCRIPTION
**Pull request by stevengill**
###  Summary

Fixes issue #1156

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: packages/socket-mode/src/SocketModeClient.ts [hunk 1]
```diff
@@ -67,22 +67,22 @@ export class SocketModeClient extends EventEmitter {
           .initialState('authenticating')
             .do(() => {
               // TODO: change this to webClient.apps.connections.open()
```
Comment (seratch) [inline packages/socket-mode/src/SocketModeClient.ts:69]:
```suggestion
```

File: packages/socket-mode/src/SocketModeClient.ts [hunk 2]
```diff
@@ -156,17 +156,18 @@ export class SocketModeClient extends EventEmitter {
             .initialState('authenticating')
               .do(() => {
                 // TODO: change this to webClient.apps.connections.open()
```
Comment (seratch) [inline packages/socket-mode/src/SocketModeClient.ts:158]:
```suggestion
```

File: packages/socket-mode/src/SocketModeClient.ts [hunk 3]
```diff
@@ -67,22 +67,22 @@ export class SocketModeClient extends EventEmitter {
           .initialState('authenticating')
             .do(() => {
               // TODO: change this to webClient.apps.connections.open()
-              return this.webClient.apiCall('apps.connections.open').then((result: WebAPICallResult) => {
+              return this.webClient.apps.connections.open().then((result: WebAPICallResult) => {
                 return result;
               }).catch((error) => {
                 this.logger.error(error);
+                // throw error;
+                return Promise.reject(error);
```
Comment (seratch) [inline packages/socket-mode/src/SocketModeClient.ts:75]: 👍

**Review by seratch — APPROVED** [review]
Although I have not yet verified if it actually works, in theory, this patch should resolve the issue. Thanks for fixing this 👍

**Review by mwbrooks — APPROVED** [review]
Thanks for getting this one all ready!

**Pull request merged**